xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java

```diff
@@ -186,201 +189,69 @@ public void tearDown() {
   }
 
   @Test
-  public void loadReportInitialRequest() {
-    verify(mockLoadReportingService).streamLoadStats(lrsResponseObserverCaptor.capture());
-    assertThat(lrsRequestObservers).hasSize(1);
-    StreamObserver<LoadStatsRequest> requestObserver = lrsRequestObservers.poll();
-    verify(requestObserver).onNext(EXPECTED_INITIAL_REQ);
-    // No more request should be sent until receiving initial response. No load reporting
-    // should be scheduled.
-    assertThat(fakeClock.getPendingTasks(LOAD_REPORTING_TASK_FILTER)).isEmpty();
-    verifyNoMoreInteractions(requestObserver);
-  }
-
-  @Test
-  public void startAndStopCanBeCalledMultipleTimes() {
-    verify(mockLoadReportingService).streamLoadStats(lrsResponseObserverCaptor.capture());
-    assertThat(lrsRequestObservers).hasSize(1);
-    StreamObserver<LoadStatsRequest> requestObserver = lrsRequestObservers.peek();
-    verify(requestObserver).onNext(EXPECTED_INITIAL_REQ);
-    lrsClient.startLoadReporting(callback);
-    assertThat(lrsRequestObservers).hasSize(1);
-    lrsClient.startLoadReporting(callback);
-    assertThat(lrsRequestObservers).hasSize(1);
-    verifyNoMoreInteractions(requestObserver);
-
-    lrsClient.stopLoadReporting();
-    assertThat(callEnded.get()).isTrue();
-    assertThat(fakeClock.getPendingTasks(LRS_RPC_RETRY_TASK_FILTER)).isEmpty();
-    lrsClient.stopLoadReporting();
-    assertThat(callEnded.get()).isTrue();
-
-    lrsClient.startLoadReporting(callback);
-    verify(mockLoadReportingService, times(2)).streamLoadStats(lrsResponseObserverCaptor.capture());
-    assertThat(lrsRequestObservers).hasSize(2);
-  }
-
-  // Currently we expect each gRPC client talks to a single service per cluster, so we test LRS
-  // client reporting load for a single cluster service only.
-  // TODO(chengyuanzhang): Existing test suites for LRS client implementation have poor behavior
-  //  coverage and are not robust. Should improve once its usage is finalized without too much
-  //  assumption.
-
-  @Test
-  public void loadReportActualIntervalAsSpecified() {
+  public void normalOperations() {
```

Comment:
The name does not provide any information what to test and what to verify. Either give it a more descriptive name or add a description of what are being tested in javadoc.

Reply:
Hmm... This indeed tests LRS client's normal operation, basically the periodic load reporting. Previously, it was splited in multiple test cases such as initial request/response, load reporting interval update, etc. It is verbose to have them be separate as usually one behavior is just a step further from another, and those test cases have significant code repeat. So I think it would be better to put them together, which is just a normal flow of LRS client. Nothing specific, normal workflow is what it is testing.

Reply:
The maybe `normalLrsFlow()` is a better name.

Reply:
`typicalWorkflow`. I saw there is a similar one in [`HealthCheckingLoadBalancerFactoryTest`](https://github.com/grpc/grpc-java/blob/659987185f9369f6d53edd2d9cfab2461409f910/services/src/test/java/io/grpc/services/HealthCheckingLoadBalancerFactoryTest.java#L251).